token_pause_transaction_e2e_test.go

```diff
@@ -0,0 +1,111 @@
+package hedera
+
+import (
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"testing"
+)
+
+func TestIntegrationTokenPauseTransactionCanExecute(t *testing.T) {
+	env := NewIntegrationTestEnv(t)
+
+	resp, err := NewTokenCreateTransaction().
+		SetNodeAccountIDs(env.NodeAccountIDs).
+		SetTokenName("ffff").
+		SetTokenSymbol("F").
+		SetAdminKey(env.Client.GetOperatorPublicKey()).
+		SetTreasuryAccountID(env.Client.GetOperatorAccountID()).
+		SetPauseKey(env.Client.GetOperatorPublicKey()).
+		SetFreezeDefault(false).
+		Execute(env.Client)
+	require.NoError(t, err)
+
+	receipt, err := resp.SetValidateStatus(true).GetReceipt(env.Client)
+	require.NoError(t, err)
+
+	tokenID := *receipt.TokenID
+
+	info, err := NewTokenInfoQuery().
+		SetTokenID(tokenID).
+		SetNodeAccountIDs([]AccountID{resp.NodeID}).
+		Execute(env.Client)
+
+	assert.NotNil(t, info.PauseStatus)
+	assert.False(t, *info.PauseStatus)
+
+	resp, err = NewTokenPauseTransaction().
+		SetNodeAccountIDs([]AccountID{resp.NodeID}).
+		SetTokenID(tokenID).
+		Execute(env.Client)
+	require.NoError(t, err)
+
+	_, err = resp.SetValidateStatus(true).GetReceipt(env.Client)
+	require.NoError(t, err)
+
+	info, err = NewTokenInfoQuery().
+		SetTokenID(tokenID).
+		SetNodeAccountIDs([]AccountID{resp.NodeID}).
+		Execute(env.Client)
+	require.NoError(t, err)
+
+	assert.NotNil(t, info.PauseStatus)
+	assert.True(t, *info.PauseStatus)
+
+	//Unpause token to avoid error in CloseIntegrationTestEnv
```

Comment:
If the test fails in some of the steps above, this wont be executed. `defer()` maybe?